loopd/swapclient_server.go

```diff
@@ -234,6 +235,42 @@ func (s *swapClientServer) Monitor(in *looprpc.MonitorRequest,
 	}
 }
 
+// GetSwaps returns a list of all currently known swaps and their current
+// status.
+func (s *swapClientServer) GetSwaps(_ context.Context,
+	_ *looprpc.GetSwapsRequest) (*looprpc.GetSwapsResponse, error) {
+
+	var (
+		rpcSwaps = make([]*looprpc.SwapStatus, len(swaps))
+		idx      = 0
+		err      error
+	)
+	for _, swp := range swaps {
+		swp := swp
+		rpcSwaps[idx], err = s.marshallSwap(&swp)
+		if err != nil {
+			return nil, err
+		}
+		idx++
+	}
+	return &looprpc.GetSwapsResponse{Swaps: rpcSwaps}, nil
+}
+
+// GetSwap returns all known details about a single swap.
+func (s *swapClientServer) GetSwap(_ context.Context,
+	req *looprpc.GetSwapRequest) (*looprpc.SwapStatus, error) {
+
+	swapHash, err := lntypes.MakeHashFromStr(req.Id)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing swap hash: %v", err)
+	}
+	swp, ok := swaps[swapHash]
```

Comment:
😞

I think this may be a good time to get rid of those global swap state. We should access the state instead directly from the swap client server, or just even read it from disk on-demand.

Reply:
I think it could be made a server level variable, and then we got get it by querying through the server.

Reply:
I removed the global swap state and went for serializing the state to the DB more often instead. This results in something like 3 updates per swap until it is completed, which is not too bad in my opinion.